### PR TITLE
Add RFC metadata as YAML frontmatter

### PR DESCRIPTION
**Pull request by richardTowers**
We reviewed the latest 42 RFCs with senior tech to check on their statuses (i.e. accepted / superseded etc.) and implementation status (i.e. done / in progress etc.).

This adds metadata to each RFC as YAML frontmatter.

Also add a script to the README to show how this can be used in a programmatic way. The results of running this script can be seen in this spreadsheet:

https://docs.google.com/spreadsheets/d/1bYF7r4IC7owbE7NrbFMZYJiuqsPZeQka1fzYD4apx9U/edit#gid=166323961

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,4 +31,31 @@ For RFCs where the outcome is an agreed Action Plan, you may want to update the
...
+<details>
+<summary>Script to list all RFC metadata</summary>
+
+```ruby
+#!/usr/bin/env ruby
+
+require "csv"
+require "yaml"
+
+frontmatter_columns = %w[status implementation status_last_reviewed status_note]
```
Comment (leenagupte) [inline README.md:47]: I think this should be `status_notes`

File: rfc-149-switch-to-per-page-asset-loading.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
 ---
-Author: Ian James
-Date: 2 August 2022
-Deadline for decision: 16 August 2022
+status: accepted
+implementation: done
+status_last_reviewed: 2024-03-04
+status_notes: >
```
Comment (leenagupte) [inline rfc-149-switch-to-per-page-asset-loading.md:5]: Will having the `>` do anything odd in the CSV?

**Review by leenagupte — COMMENTED** [review]
Looks good. Once minor inline fix to get the status notes to show in the CSV and one small question.

Your title of Spreadsheet King remain unchallenged! 👑

File: rfc-149-switch-to-per-page-asset-loading.md (hunk 2 above)
Comment (richardTowers) [reply rfc-149-switch-to-per-page-asset-loading.md:5]: No, it's YAML syntax for a Folded Block Scalar String. So long as it's parsed properly as YAML, the `>` character won't appear in the output. More info on YAML's disgustingly overkill multiline string handling: https://yaml-multiline.info/

**Review by leenagupte — APPROVED** [review]

File: rfc-165-add-api-endpoints-to-transition.md [hunk 3]
```diff
@@ -1,3 +1,12 @@
+---
+status: to_be_deleted
+implementation: not_planned
+status_last_reviewed: 2024-03-04
+status_notes: >
+  People aren't clear whether the value of doing this outweighs the cost of implementing it.
```
Comment (JonathanHallam) [inline rfc-165-add-api-endpoints-to-transition.md:6]: I'm still not 100% on this, I think i'd still like to at least spike it but I'll have a chat with the team :)

**Pull request merged**